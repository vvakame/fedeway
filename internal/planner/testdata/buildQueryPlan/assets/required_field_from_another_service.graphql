# schema: supergraphSdl.graphqls

# when requesting a field defined in another service which requires a field in the base service
# should add the field provided by base service in first Fetch
query {
    topCars {
        retailPrice
    }
}
