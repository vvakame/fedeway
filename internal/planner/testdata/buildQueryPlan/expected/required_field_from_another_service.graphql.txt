QueryPlan {
	Sequence {
		Fetch(service: "product") {
			query {
				topCars {
					__typename
					id
					price
				}
			}
		},
		Flatten(path: "topCars.@") {
			Fetch(service: "reviews") {
				{
					... on Car {
						__typename
						id
						price
					}
				} =>
				{
					... on Car {
						retailPrice
					}
				}
			},
		},
	},
}